controllers/datafoundation.go

```diff
@@ -19,32 +30,53 @@ type dataFoundationSpec struct {
 type dataFoundationReconciler struct {
 	*ManagedFusionOfferingReconciler
 
+	offering                      *v1alpha1.ManagedFusionOffering
 	spec                          dataFoundationSpec
 	onboardingValidationKeySecret *corev1.Secret
 }
 
+//+kubebuilder:rbac:groups=operators.coreos.com,namespace=system,resources=clusterserviceversions,verbs=get;list;watch;delete;update;patch
+
 func dfSetupWatches(controllerBuilder *builder.Builder) {
+	csvPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(client client.Object) bool {
+				return strings.HasPrefix(client.GetName(), ocsOperatorName)
+			},
+		),
+	)
+	enqueueByAnnotaion := utils.NewEnqueueByAnnotation(managedFusionAnnotation)
 	controllerBuilder.
-		Owns(&corev1.Secret{})
+		Owns(&corev1.Secret{}).
+		Watches(
+			&source.Kind{Type: &opv1a1.ClusterServiceVersion{}},
+			enqueueByAnnotaion,
+			csvPredicates,
+		)
 }
 
 func dfReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) error {
 	r := dataFoundationReconciler{}
-	r.initReconciler(offeringReconciler)
+	r.initReconciler(offeringReconciler, offering)
 
 	if err := r.parseSpec(offering); err != nil {
 		return err
 	}
 	if err := r.reconcileOnboardingValidationSecret(); err != nil {
 		return err
 	}
+	if err := r.reconcileCSV(); err != nil {
+		return err
+	}
 
 	return nil
 }
 
-func (r *dataFoundationReconciler) initReconciler(offeringReconciler *ManagedFusionOfferingReconciler) {
+func (r *dataFoundationReconciler) initReconciler(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) {
 	r.ManagedFusionOfferingReconciler = offeringReconciler
 
+	r.offering = offering
```

Comment:
If we only care about the Namespaced Name of the offering, let's just save that. or even just the name and namespace separately. And to think about it you already have access to the namespace via r.Namespace. So maybe it is better to make r.Namespace private and get it from the offering resource
 ```suggestion
	r.offeringNamespace = offering.Namespace
	r.offeringName = offeringName
	
```

Reply:
Updated